lib/internal/validators.js

```diff
@@ -126,6 +127,20 @@ function validateNumber(value, name) {
     throw new ERR_INVALID_ARG_TYPE(name, 'number', value);
 }
 
+const validateOneOf = hideStackFrames((value, name, oneOf, option = false) => {
+  if (!oneOf.includes(value)) {
+    const allowed = oneOf
+      .map((v) => (typeof v === 'string' ? `'${v}'` : String(v)))
+      .join(', ');
+    const reason = 'must be one of: ' + allowed;
+    if (!option) {
+      throw new ERR_INVALID_ARG_VALUE(name, value, reason);
+    } else {
+      throw new ERR_INVALID_OPT_VALUE(name, value, reason);
+    }
```

Comment:
FWIW, I generally ignore `ERR_INVALID_OPT_VALUE` and instead use the `option.${name}` pattern instead with `ERR_INVALID_ARG_VALUE`

Reply:
I'd love to see that. I always found different error codes for OPT and ARG weird since they are not really useful imo.
I will work on it :smile: .